rebasehelper/application.py

```diff
@@ -490,113 +490,143 @@ def build_packages(self):
-
-                if e.logfile is None:
-                    msg = 'Building {} SRPM packages failed; see logs in {} for more information'.format(
-                        version, os.path.join(results_dir, 'SRPM')
-                    )
-                else:
-                    msg = 'Building {} SRPM packages failed; see {} for more information'.format(version, e.logfile)
-                raise RebaseHelperError(msg, logfiles=builder.get_logs().get('logs'))
-
-            except BinaryPackageBuildError as e:
-                build_dict.update(builder.get_logs())
-                build_dict['binary_package_build_error'] = six.text_type(e)
-                results_store.set_build_data(version, build_dict)
-
-                if e.return_code:
-                    # koji was used for build therefore only status code is set
-                    e.logfile = MockBuildTool.get_mock_logfile_path(e.return_code,
-                                                                    os.path.join(results_dir, 'RPM'))
-                if e.logfile:
-                    msg = 'Building {} RPM packages failed; see {} for more information'.format(version, e.logfile)
-                else:
-                    msg = 'Building {} RPM packages failed; see logs in {} for more information'.format(
-                        version, os.path.join(results_dir, 'RPM')
-                    )
-                raise RebaseHelperError(msg, logfiles=builder.get_logs().get('logs'))
-
-            except Exception as e:
-                raise RebaseHelperError('Building package failed with unknown reason. '
-                                        'The error reported is "{}". '
-                                        'Check all available log files.'.format(six.text_type(e)))
+                        msg = 'Building {} SRPM packages failed; see {} for more information'.format(version, e.logfile)
+                    raise RebaseHelperError(msg, logfiles=builder.get_logs().get('logs'))
+
+                except BinaryPackageBuildError as e:
+                    build_dict.update(builder.get_logs())
+                    build_dict['binary_package_build_error'] = six.text_type(e)
+                    results_store.set_build_data(version, build_dict)
+
+                    if e.logfile is None:
+                        msg = 'Building {} RPM packages failed; see logs in {} for more information'.format(
+                            version, os.path.join(results_dir, 'RPM')
+                        )
+                    else:
+                        msg = 'Building {} RPM packages failed; see {} for more information'.format(version, e.logfile)
+
+                    logger.info(msg)
+                    if not self.conf.non_interactive and \
+                            ConsoleHelper.get_message('Do you want to try it one more time?'
+                                                      ' (You will be asked to change %s)' % rebased_spec_path):
```

Comment:
> You will be asked to change ...

This doesn't seem right, because user is not asked that actually. I'd rather see something like this (the wording could definitely be improved):

```
...
Do you want to try it one more time? [Y/n] y
Now it's time to modify rebase-helper-results/rebased-sources/foo.spec if necessary.
Do you want to continue with the rebuild now? [Y/n] y
...
```

Reply:
I was thinking about the wording, and this seems most suitable. Thanks!